module.go

```diff
@@ -226,6 +226,7 @@ func (m *Module) inspectorPreRequest(req *http.Request) (inspin2 RPCMsgIn2, out
 	}
 
 	inspin := NewRPCMsgIn(req, reqbody, -1, -1, -1, m.config.ModuleIdentifier(), m.config.ServerIdentifier())
+	inspin.ServerFlavor = m.config.ServerFlavor()
```

Comment:
Nice idea to not change the API. I think there will be an issue here, though.

See: https://github.com/signalsciences/sigsci-agent/blob/b832b73fc0d9638f7f371c75ad0ce4205316a1c1/revproxy/revproxy.go#L518-L522

That would also have to be modified to do as you did here and update the `ServerFlavor`. Instead, I think make another constructor:

```go
func NewRPCMsgInWithModuleConfig(mcfg *ModuleConfig, r *http.Request, postbody []byte, code int, size int64, dur time.Duration) *RPCMsgIn
```

And then just do:
```go
inspin := NewRPCMsgInWithModuleConfig(m.config, req, reqbody, -1, -1, -1)
```

Reply:
Might even make some sense to just pair it down to just:

```go
func NewRPCMsgInWithModuleConfig(mcfg *ModuleConfig, r *http.Request, postbody []byte) *RPCMsgIn
```

And if adding to the API like this, should it return an error as well?

Reply:
ok, I added the new constructor, lemme know what you think.